fix(performance-views) - Return nothing instead of 404

wmak opened the pull request:
- When there are no key transactions for a user return nothing with a
  200 instead of a 404

wmak committed:
fix(performance-views) - Return nothing instead of 404

- When there are no key transactions for a user return nothing with a
  200 instead of a 404

dashed commented on src/sentry/discover/endpoints/discover_key_transactions.py:
```diff
@@ -110,7 +114,7 @@ def get(self, request, organization):
         queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
         if not queryset.exists():
-            raise ResourceDoesNotExist
+            return Response({"data": []}, status=200)
```
@wmak this may not work if there are multiple y-axis

wmak replied:
@dashed Ended up zerofilling the results cause that would be easier, let me know if that's an issue though and I can update.

wmak committed:
ref: Support multi yaxis when no queryset

- This means that we'll still zerofill results, but no snuba query is
  being made without a queryset

wmak commented on src/sentry/discover/endpoints/discover_key_transactions.py:
```diff
@@ -68,12 +68,16 @@ def get(self, request, organization):
-            fields, request.GET.get("query"), params, orderby, "discover.key_transactions", queryset
-        )
+        results = {}
+        if queryset.exists():
```
Will do a latter refactor to move this check into the query function, not being done here because I need to remove the conditions param from `discover.query` and cleaner to do those two together.

dashed reviewed (approved):
Looks good to me 👍
